src/sqlfluff/core/rules/std/L045.py

```diff
@@ -0,0 +1,73 @@
+"""Implementation of Rule L045."""
+from typing import Dict, List
+
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.rules.base import BaseCrawler, LintResult
+from sqlfluff.core.rules.analysis.select_crawler import SelectCrawler
+from sqlfluff.core.rules.doc_decorators import document_fix_compatible
+
+
+@document_fix_compatible
+class Rule_L045(BaseCrawler):
+    """Query defines a CTE (common-table expression) but does not use it.
+
+    | **Anti-pattern**
+    | Defining a CTE that is not used by the query is harmless, but it means
+    | the code is unnecesary and could be removed.
```

Comment:
very cool that we're able to detect that :)

Reply:
It was a good exercise -- I was really happy to see the code was so simple by reusing code from the other rule. I think if we keep refactoring the reusable code, it'll be really great in a few months. For now it works, which is a good start. 😄